chronos/asyncfutures2.nim

```diff
@@ -55,10 +60,15 @@ type
   # Obviously, it will still be allocated on the heap when necessary.
   Future*[T] = ref object of FutureBase ## Typed future.
     when chronosStrictException:
-      closure*: iterator(f: Future[T]): FutureBase {.raises: [Defect, CatchableError], gcsafe.}
+      when (NimMajor, NimMinor) < (1, 4):
+        closure*: iterator(f: Future[T]): FutureBase {.raises: [Defect, CatchableError], gcsafe.}
+      else:
+        closure*: iterator(f: Future[T]): FutureBase {.raises: [CatchableError], gcsafe.}
     else:
-      closure*: iterator(f: Future[T]): FutureBase {.raises: [Defect, CatchableError, Exception], gcsafe.}
-    value: T ## Stored value
+      closure*: iterator(f: Future[T]): FutureBase {.raises: [Exception], gcsafe.}
+
+    when T isnot void:
+      value*: T ## Stored value
 
   FutureStr*[T] = ref object of Future[T]
```

Comment:
This could be deprecated.

Reply:
I agree but it's beyond the scope of this PR: they are widely used in chronos itself so deprecating them here and now would result in a lot of warnings.